File: app/src/main/java/alireza/nezami/dependency_injection_without_dagger/java/approach1/ui/MainViewModel.java
package alireza.nezami.dependency_injection_without_dagger.java.approach1.ui;

import android.content.Context;

import alireza.nezami.dependency_injection_without_dagger.java.approach1.data.local.DatabaseService;
import alireza.nezami.dependency_injection_without_dagger.java.approach1.data.remote.NetworkService;


public class MainViewModel {

    private DatabaseService databaseService;
    private NetworkService networkService;

    public MainViewModel(Context context) {
        networkService = new NetworkService(context);
        databaseService = new DatabaseService(context);
    }

    public String getSomeData() {
        return databaseService.getDummyData() + " : " + networkService.getDummyData();
    }
}

File: app/src/main/java/alireza/nezami/dependency_injection_without_dagger/kotlin/approach1/ui/MainViewModel.kt
package alireza.nezami.dependency_injection_without_dagger.kotlin.approach1.ui

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import alireza.nezami.dependency_injection_without_dagger.R
import alireza.nezami.dependency_injection_without_dagger.kotlin.approach1.data.local.DatabaseService
import alireza.nezami.dependency_injection_without_dagger.kotlin.approach1.data.remote.NetworkService
import android.content.Context
import android.widget.TextView

class MainViewModel(context: Context?) {
    private val databaseService: DatabaseService = DatabaseService(context)
    private val networkService: NetworkService = NetworkService(context)
    val someData: String
        get() = databaseService.dummyData + " : " + networkService.dummyData
}